lib/Service/ProviderService.php

```diff
@@ -412,33 +412,44 @@ private function login($uid, Profile $profile, $newGroupPrefix = '')
             }
 
             if (isset($profile->data['groups']) && is_array($profile->data['groups'])) {
-                $groupNames = $profile->data['groups'];
+                $groups = $profile->data['groups'];
                 $groupMapping = isset($profile->data['group_mapping']) ? $profile->data['group_mapping'] : null;
                 $userGroups = $this->groupManager->getUserGroups($user);
                 $autoCreateGroups = $this->config->getAppValue($this->appName, 'auto_create_groups');
-                $syncGroupNames = [];
+                $syncGroups = [];
 
-                foreach ($groupNames as $k => $v) {
-                    if ($groupMapping && isset($groupMapping[$v])) {
-                        $syncGroupNames[] = $groupMapping[$v];
+                foreach ($groups as $k => $v) {
+                    if(is_object($v)) {
+                        $group = $v;
```

Comment:
```suggestion
                        if (empty($v->gid)) {
                            continue;
                        }
                        $group = $v;
```
As input may be wrong

Reply:
Wouldn't it be better to use `!isset()`, since an `gid` of 0 will now also be ignored.

Reply:
What about empty string ?

Reply:
Good point, I think `gid` 0 should probably never happen, so then `empty` would be indeed a better idea